### PR TITLE
#77 - Authentication method issue fixed

### PR DESCRIPTION
**Pull request by salmanahmed-postgresql**

### CODE REVIEW

File: 3-node-cluster/ansible/playbooks/setup-pgpool.yml [hunk 1]
```diff
@@ -54,6 +62,15 @@
           backend_data_directory2 = '{{ PG_CIRRUS_INSTALLATION_DIRECTORY }}' # Data directory path for standby2 PostgreSQL server.
         insertafter: EOF
 
+    # Create a pgpool password file in /etc/pgpool2
```
Comment (DevBotX) [inline 3-node-cluster/ansible/playbooks/setup-pgpool.yml:65]: Are we setting the proper permissions for this file as well? i.e, 
Allowing only the postgres user for read/write operations

File: 3-node-cluster/pgpool-internals/follow_primary.sh [hunk 2]
```diff
@@ -45,7 +45,7 @@ fi
 ssh postgres@$NEW_MAIN_NODE_HOST "psql -d postgres -w -c \"SELECT pg_create_physical_replication_slot('slot_$SLOT_NAME');\""
 
 # Update connection string on all standby nodes to point to new primary
-ssh postgres@$NODE_HOST "psql -d postgres -w -c \"ALTER SYSTEM SET primary_conninfo = 'user=repuser host=$NEW_MAIN_NODE_HOST port=$NODE_PORT';\""
+ssh postgres@$NODE_HOST "PGPASSFILE=~/.pgpass psql -d postgres -w --set=sslmode=disable -c \"ALTER SYSTEM SET primary_conninfo = 'user=repuser host=$NEW_MAIN_NODE_HOST port=$NODE_PORT';\""
```
Comment (DevBotX) [inline 3-node-cluster/pgpool-internals/follow_primary.sh:48]: I have some confusion about --set=sslmode=disable is syntax correct and if yes why we are disabling ssl-mode?

File: 3-node-cluster/ansible/playbooks/setup-pgpool.yml (hunk 1 above)
Comment (salmanahmed-postgresql) [reply 3-node-cluster/ansible/playbooks/setup-pgpool.yml:65]: We are using this task to write to pool_passwd file which I believe gets created when pgpool is installed
I think I should update the commend and name of the task

    # Create a pgpool password file in /etc/pgpool2
    - name: Create a pool_passwd file
      lineinfile:
        path: /etc/pgpool2/pool_passwd
        line: |
          repuser:{{ REPUSER_PASSWORD }}
          postgres:{{ PFILE_PASSWORD }}
        insertafter: EOF

File: 3-node-cluster/pgpool-internals/follow_primary.sh (hunk 2 above)
Comment (salmanahmed-postgresql) [reply 3-node-cluster/pgpool-internals/follow_primary.sh:48]: You are right this was not in out initial code, it must have been added mistakenly while editing this, I'll remove it

File: 3-node-cluster/ansible/playbooks/setup-pgpool.yml (hunk 1 above)
Comment (salmanahmed-postgresql) [reply 3-node-cluster/ansible/playbooks/setup-pgpool.yml:75]: The permissions of pool_passwd file are as following:

-rw-r----- 1 root postgres   30 Aug  1 12:08 pool_passwd

**Review by DevBotX — APPROVED** [review]

**Pull request merged**